### DISCUSSION TITLE
Export view as Query

### DISCUSSION THREAD

**Discussion by nodeGarden** [directus/directus#23033] [Feature Requests]
### Describe the Request

It would be very useful to take what is shown inside a data table, such as the Content section viewer, or Insights component, and export a fully qualified API query for what rules are populating it.

Say I have applied a filter to the results of the Content > Collection view, and I want to make the same available in another part of the UI (like Insights) or Externally. It would be great if we could get a CURL/Fetch/SQL/etc that will give us the same results. Bonus points if we can CHOOSE between multiple formats.

![image](https://github.com/directus/directus/assets/253262/4ccbb938-d1e8-4a69-9f43-88aa8c741832)

The same for Insights, because some of the components might have their own filters/aggregations, it would be great to create a "validation" query to go direct to the SQL DB and confirm everything is displaying correctly.

### Maintainence Strategy

Would need to be kept up-to-date with all filter updates

**Comment by shipsides-sparknel** [comment]
i do something similar using a saved preset.

in Directus, i created a custom preset on my collection that has all of my filtering logic and fields, and noted the preset ID it creates. when i want to access the preset programmatically, i make a call to the /presets endpoint for .data.filter to get the JSON, and .data.layout_query.tabular.fields then join with commas to get the field list as text.

then i hit the collection items api with filter=presetFilter&fields=presetFields. the benefit of this is that the query remained synced both inside and outside the app.

i love the simplicity of OP's proposal and see the value, but i can't help but think that a way to directly call a preset via the items endpoint to essentially create a linked query is the "better" way.
